uniqueidentifier_null.go

```diff
@@ -0,0 +1,49 @@
+package mssql
+
+import (
+	"database/sql/driver"
+)
+
+type NullUniqueIdentifier struct {
+	UUID  UniqueIdentifier
+	Valid bool // Valid is true if UUID is not NULL
+}
+
+func (n *NullUniqueIdentifier) Scan(v interface{}) error {
+	if v == nil {
+		*n = NullUniqueIdentifier{
+			UUID:  [16]byte{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0},
+			Valid: false,
+		}
+		return nil
+	}
+	u := n.UUID
+	err := u.Scan(v)
+	*n = NullUniqueIdentifier{
+		UUID:  u,
+		Valid: true,
+	}
+	return err
+}
+
+func (n NullUniqueIdentifier) Value() (driver.Value, error) {
+	return n.UUID.Value()
```

Comment:
>turn n.UUID.Value() [](http://example.com/codeflow?start=3&length=19)

Won't Value/String/Marshal etc need to handle `!Valid`?
Pls add tests for these. If there are existing functional tests that cover `String()` and `Value()`  implementations of other types let's add a case to those too.

Reply:
I wasn´t entirely sure what `!Valid` for String should be 34e68bcc970ab6f7b3142f499f6d3e3d51b117f8

Reply:
does this type need to implement `String()` ?
I don't see the core nullable types implementing it 
[https://cs.opensource.google/go/go/+/master:src/database/sql/sql.go?q=nullint64&ss=go%2Fgo](https://cs.opensource.google/go/go/+/master:src/database/sql/sql.go?q=nullint64&ss=go%2Fgo)